logdetective/logdetective.py

```diff
@@ -49,8 +49,13 @@ def main():
     LOG.setLevel(log_level)
 
     # Primary model initialization
-    model = initialize_model(args.model, filename_suffix=args.filename_suffix,
-                             verbose=args.verbose > 2)
+    try:
+        model = initialize_model(args.model, filename_suffix=args.filename_suffix,
+                                 verbose=args.verbose > 2)
+    except ValueError as e:
```

Comment:
Do you think we could match on the `failed to allocate buffer` string somehow? I suppose it ends up in stderr. Otherwise we could go for `Failed to create llama_context`, so we catch only this specific error..

Reply:
It is hard for me to reproduce it. So I would prefer to keep it as it is.

Reply:
Ok. That makes sense. We can always come back to this later.